### DISCUSSION TITLE
A single (proper) bundle? Make it work on Deno?

### DISCUSSION THREAD

**Discussion by tunnckoCore** [wevm/viem#199] [Idea / Feature Request] [answered]
Heya there.

Fellow veteran Open Sourcer playing here, with this shiny new thing.

I almost give up and went to build my app on the client side, or even with just basic `window.ethereum`, but then you release this one. I thought, okay, this is small and looks good, and went again to Deno Fresh.

Don't know if you are familiar with Deno, or Fresh in particular, but Fresh has Islands (client side components) like Astro. Thought, okay, I'll import viem from `esm.sh` and create a client side component. Not much luck. It seems like it is very splitted and depends on many other external deps, I included them in the import-map too - nope. Tried the ESM exports from UNPKG too.. Can't make it load properly without errors. The most common and weird one was around the `constants` file about Deno. Not even sure if it was from the `viem` or from some of its deps. Also seems like that esm.sh doesn't properly making its bundle additionally giving some errors like 

```
TypeError: Relative import path "isomorphic-ws" not prefixed with / or ./ or ../ and not in import map from "https://unpkg.com/viem@0.1.13/dist/chunk-S4MOKUQW.mjs"
```

Trying skypack's cdn now...

Nope, they definitely don't like something about the chunking you're doing.

```
error: Uncaught (in promise) TypeError: Relative import path "isomorphic-ws" not prefixed with / or ./ or ../ and not in import map from "https://cdn.skypack.dev/-/viem@v0.1.13-n8XbKo1u9ijGW5SL9YZp/dist=es2019,mode=imports/optimized/common/chunk-3FWCHYKE-f6488b97.js"
    at https://cdn.skypack.dev/-/viem@v0.1.13-n8XbKo1u9ijGW5SL9YZp/dist=es2019,mode=imports/optimized/common/chunk-3FWCHYKE-f6488b97.js:3:27
  await import(entrypoint);
```

Raw from Github typescript source (rawgit cdn which is now at JSDelivr) doesn't work either, because does not have extensions in the import specifiers... 

So, I'm thinking if you can provide a single (and minified) ESM bundle of the internal things (and eventually the external ones)?

Btw, you have typo in the `main` branch's package.json `"module": "./dist/index.mts",`

**Comment by jxom** [comment, marked as answer]
Can have a look into it! First-class support for Deno would definitely be nice.

> Btw, you have typo in the main branch's package.json "module": "./dist/index.mts"

This is expected (we use `.mts` in dev) – before we publish, we convert this to `.mjs` (you can see this [here](https://www.runpkg.com/?viem@0.1.13/package.json)).

**Reply by tunnckoCore** [reply]
Awesome. I'll go with window.ethereum for now :laughing: 

> This is expected (we use .mts in dev) – before we publish, we convert this to .mjs (you can see this [here](https://www.runpkg.com/?viem@0.1.13/package.json)).

Thought about that, cuz seen the cdn package is ok.
